src/object.c

```diff
@@ -333,17 +333,7 @@ void freeZsetObject(robj *o) {
 }
 
 void freeHashObject(robj *o) {
-    switch (o->encoding) {
-    case OBJ_ENCODING_HT:
-        dictRelease((dict*) o->ptr);
-        break;
-    case OBJ_ENCODING_LISTPACK:
-        lpFree(o->ptr);
-        break;
-    default:
-        serverPanic("Unknown hash encoding type");
-        break;
-    }
+    hashTypeFree(o);
 }
```

Comment:
Not sure I understand. For listpack, there is no private ebuckets. 
Is this related about deleting object without deleting it from global ebuckets?

Reply:
I know that. I just try to say that you don't see here any footprint of HFE is because it is hidden in the call dictRelease(). 
You do need to release object OBJ_ENCODING_LISTPACK_EX.

Reply:
but hashTypeFree() releases OBJ_ENCODING_LISTPACK_EX as well.

Reply:
My mistake. I was looking on obsolete code.